pkg/handler/v2/application_installation/application_installation.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applicationinstallation
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/endpoint"
+
+	apiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	apiv2 "k8c.io/kubermatic/v2/pkg/api/v2"
+	appskubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/apps.kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/handler/middleware"
+	"k8c.io/kubermatic/v2/pkg/handler/v1/common"
+	"k8c.io/kubermatic/v2/pkg/provider"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ListApplicationInstallations(userInfoGetter provider.UserInfoGetter) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(listApplicationInstallationsReq)
+
+		client, err := userClientFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		installList := &appskubermaticv1.ApplicationInstallationList{}
+		if err := client.List(ctx, installList); err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		installations := make([]*apiv2.ApplicationInstallation, len(installList.Items))
+		for i := range installList.Items {
+			installations[i] = convertInternalToExternal(&installList.Items[i])
+		}
+
+		return installations, nil
+	}
+}
+
+func CreateApplicationInstallation(userInfoGetter provider.UserInfoGetter) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(createApplicationInstallationReq)
+
+		client, err := userClientFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
```

Comment:
nit: I would prefer to be more explicit about the fact that we use user cluster client here (and in other functions as well). E.g: 
```suggestion
		userClusterClient, err := userClientFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
```
```suggestion
		client, err := userClusterClientFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
```